graph/telemetry/istio/appender/service_entry.go

```diff
@@ -54,20 +67,51 @@ func (a ServiceEntryAppender) applyServiceEntries(trafficMap graph.TrafficMap, g
 			continue
 		}
 
-		// A service node with no outgoing edges may be an egress.
-		// If so flag it, don't discard it (kiali-1526, see also kiali-2014).
-		// The flag will be passed to the UI to inhibit links to non-existent detail pages.
-		if location, ok := a.getServiceEntry(n.Service, globalInfo); ok {
-			n.Metadata[graph.IsServiceEntry] = location
+		// A service node with no outgoing edges represents a serviceEntry when the service name matches
+		// serviceEntry host. Map these "se" nodes to the serviceEntries that represent them.
+		if se, ok := a.getServiceEntry(n.Service, globalInfo); ok {
+			if nodes, ok := seMap[se]; ok {
+				seMap[se] = append(nodes, n)
```

Comment:
I think something is failing with this mapping, I can see the following in the logs
```
2019-06-21T16:50:36.358202391Z W0621 16:50:36.358145       1 service_entry.go:83] Aggregating into {location:MESH_EXTERNAL name:external-svc-apple}
2019-06-21T16:50:36.358209553Z W0621 16:50:36.358154       1 protocol.go:223] In AggregateNodemetadata...
2019-06-21T16:50:36.358214627Z W0621 16:50:36.358168       1 protocol.go:227] protocol=tcp: rate=tcpIn: add 24142.048.2
2019-06-21T16:50:36.358220445Z W0621 16:50:36.358195       1 service_entry.go:102] Deleting {ID:svc_travel-agency_www.apple.com NodeType:service Namespace:travel-agency Workload: App: Version: Service:www.apple.com Edges:[] Metadata:map[destServices:map[travel-agency www.apple.com:{Namespace:travel-agency Name:www.apple.com}] tcpIn:24142.048]}
2019-06-21T16:50:36.35825145Z W0621 16:50:36.358218       1 service_entry.go:105] Adding {ID:svc_travel-agency_external-svc-apple NodeType:service Namespace:travel-agency Workload: App: Version: Service:external-svc-apple Edges:[] Metadata:map[isServiceEntry:MESH_EXTERNAL destServices:map[travel-agency www.apple.com:{Namespace:travel-agency Name:www.apple.com}] tcpIn:24142.048]}
2019-06-21T16:50:36.358263516Z W0621 16:50:36.358230       1 service_entry.go:83] Aggregating into {location:MESH_EXTERNAL name:external-svc-apple}
2019-06-21T16:50:36.358277446Z W0621 16:50:36.358241       1 protocol.go:223] In AggregateNodemetadata...
2019-06-21T16:50:36.358283315Z W0621 16:50:36.358253       1 protocol.go:227] protocol=tcp: rate=tcpIn: add 22109.291.2
2019-06-21T16:50:36.35830052Z W0621 16:50:36.358273       1 service_entry.go:102] Deleting {ID:svc_travel-agency_www.ishopmixup.com NodeType:service Namespace:travel-agency Workload: App: Version: Service:www.ishopmixup.com Edges:[] Metadata:map[destServices:map[travel-agency www.ishopmixup.com:{Namespace:travel-agency Name:www.ishopmixup.com}] tcpIn:22109.291]}
2019-06-21T16:50:36.358319124Z W0621 16:50:36.358295       1 service_entry.go:105] Adding {ID:svc_travel-agency_external-svc-apple NodeType:service Namespace:travel-agency Workload: App: Version: Service:external-svc-apple Edges:[] Metadata:map[isServiceEntry:MESH_EXTERNAL destServices:map[travel-agency www.ishopmixup.com:{Namespace:travel-agency Name:www.ishopmixup.com}] tcpIn:22109.291]}
```

I think that the reason is because different ServiceEntries are being created here for each host: https://github.com/kiali/kiali/pull/1172/files#diff-e08a9c750150eee4d24831ee0f9bc0b2R134 (line 134 of this file)